fcgl/user-interaction
# Authentication workflow

## Problem
We need a way to authenticate our users. This implementation must allow other services to not have to sign in to use each api

## Solution
* When user signs in. They will be grated a JWT token. This token needs to be passed in the header for every request in order for it to be valid. As long as the JWT token is valid the user has access to their account.

* Add `/signup` and `/login` endpoints

* Add Oatuh2.0 implementation for google. It only works on web... need to look into how to do it for react native

## Test
1. Run and build: `docker-compose up --build`
2. Confirm that you receive a JWT token when login in
